pages/Proxy-Basics.md

```diff
@@ -6,15 +6,34 @@ nav_order: 2
 
-## What is a proxy and why are they useful?
+## Why do I need a proxy and how do I use it?
 
-Contract code on a blockchain is immutable, meaning it cannot be altered. This is a very useful property, one of the key benefits of blockchains. But code that cannot be changed also has the downside of being hard or impossible to update. Updates can be good, whether they provide users with new features or fix bugs in the original code. Proxies provide a way for smart contract code to be updated. There are two main approaches for proxies:
+By design, contract code on a blockchain is **immutable**. Though a key feature, it leads to difficulty when considering upgradeability. Newer entrants may wonder why "upgrading" on a blockchain is necessary. Inevitabilities requiring code changes still remain, including: bug fixes, patches, optimizations, feature releases, etc.
 
-1. A lightweight "entry point" contract with a `delegatecall` that uses the logic of another contract. The first contract stores the address of the logic contract in a variable that can be changed, so when a new logic contract is deployed at a new address, the "entry point" can remain unchanged because only a state variable used for the `delegatecall` operation needs changing.
-2. A factory contract creates the logic contract with CREATE2. The logic contract contains a `selfdestruct` (normally protected so only an admin or authorized account can call it) to allow the contract to be destroyed. When the contract is destroyed, the factory can use CREATE2 to deploy a new contract with different code at the same address as the old destroyed contract.
+The **Proxy** or **Proxy Delegate** pattern allows for this upgradeability. Put simply, there are two main approaches for proxies:
+
+1. A lightweight, entry-point caller contract `A` to use the logic of a callee contract `B` through `delegatecall()`:
+- `A` stores the address of `B` in a state variable that can be changed. 
+- If an upgrade is desired, a new callee contract `C` is deployed at a different address.
+- `A` updates the state variable to point to the new callee contract, `C`.
+
+2. A factory contract `A` that creates a logic contract `B` with opcode `CREATE2`:
+- `B` contains a `selfdestruct()` to allow the contract to be destroyed. 
+  - *Note: This is normally protected so only an admin or authorized account can call it.*
+- If an upgrade is desired, `B` is destroyed, and the factory can use `CREATE2` to deploy a new contract at the same address as `B`.
+
+Additional reading for the proxy pattern can be found [here](https://docs.openzeppelin.com/upgrades-plugins/1.x/proxies), and [here](https://fravoll.github.io/solidity-patterns/proxy_delegate.html).
 
 ## Can smart contracts be upgraded without proxies?
```

Comment:
I feel like this section is interesting enough to be in its own page in the future